### ISSUE TITLE
val.py --task speed parser parameter return nothing

### ISSUE THREAD

**Issue by Haebuk** [ultralytics/yolov5#5600]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Validation

### Bug

When excuting val.py with --task speed, the --name parser parameter is ignored and return nothing.

```bash
$python yolov5/val.py --data VisDrone.yaml --name bug-testing --weights YOLOv5/yolov5n/weights/best.pt --task speed

...

Results saved to yolov5/runs/val/exp5
```

### Environment

- YOLO: YOLOv5 🚀 v6.0-74-gb8f979b torch 1.10.0+cu113 CUDA:0 (NVIDIA GeForce RTX 3080 Ti, 12051MiB)
- OS: Ubuntu 20.
- Python: 3.8.12

### Minimal Reproducible Example

```bash
python val.py --data VisDrone.yaml --task speed --name bug-test
```

```bash

> Result saved to yolov5/runs/val/exp (not bug-test, and there is no file in the folder)
```

### Additional

When excuting the shell without --task parser, It returns well.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Haebuk thanks for the bug report! --project and --name should always be used to create your save_dir = project/name. I'll take a look.

**Comment by glenn-jocher** [comment]
@Haebuk good news 😃! Your original issue may now be fixed ✅ in PR #5600. This PR updated val.py to pass all arguments by default to all tasks. Note though that --task speed will not produce any saved output, the results are simply displayed to screen. 

There was a separate PR https://github.com/ultralytics/yolov5/pull/5341 which attempted to log console outputs to text file also, but this was never merged.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
